CMakeLists.txt

```diff
@@ -157,6 +155,9 @@ option(BUILD_SHARED_LIBS_WITH_STATIC_MONGOC
     OFF
 )
 
+# Allow the user to opt into using bsoncxx implementations for C++17 polyfills by default.
+option(ENABLE_BSONCXX_POLY_USE_IMPLS "Enable using bsoncxx implementations of C++17 polyfills for pre-C++17 configurations by default" OFF)
```

Comment:
I'm iffy on this CMake setting name. It is unclear what it means compared to `USE_BOOST`, `USE_STD`, `USE_MNMLSTC` etc. I'm unfortunately not sure on what a better name might be.